pkg/stores/sqlproxy/proxy_store.go

```diff
@@ -803,6 +811,7 @@ func (s *Store) ListByPartitions(apiOp *types.APIRequest, apiSchema *types.APISc
 	if err != nil {
 		return nil, 0, "", fmt.Errorf("cachefor %v: %w", gvk, err)
 	}
+	defer s.cacheFactory.DoneWithCache(inf)
```

Comment:
I assume this is in a `defer` to handle the early-return conditions if an error is detected before getting to `inf.ListByOptions()`, right? Because we can free the lock immediately after that call, right? I guess once we're there we aren't far from returning from the function, so might as well stick with the `defer(...)`.

Reply:
Right, as soon as `CacheFor` succeeds, we need to ensure that `DoneWithCache` is called even if the rest of this function returns an error.